osu.Game/Screens/Edit/Editor.cs

```diff
@@ -849,6 +849,40 @@ private void exportBeatmap()
             new LegacyBeatmapExporter(storage).Export(Beatmap.Value.BeatmapSetInfo);
         }
 
+        private void deleteDifficulty()
+        {
+            dialogOverlay?.Push(new PromptForDifficultyDeleteDialog(confirmDifficultyHide, confirmDifficultyDelete, () => { }));
+        }
+
+        private void confirmDifficultyHide()
+        {
+            var current = playableBeatmap.BeatmapInfo;
+            if (current is null) return;
+
+            beatmapManager.Hide(current);
+            switchBeatmapOrExit(current.BeatmapSet);
+        }
+
+        private void confirmDifficultyDelete()
+        {
+            var current = playableBeatmap.BeatmapInfo;
+            if (current is null) return;
+
+            beatmapManager.DeleteLocal(current);
+            switchBeatmapOrExit(current.BeatmapSet);
+        }
+
+        private void switchBeatmapOrExit([CanBeNull] BeatmapSetInfo setInfo)
+        {
+            if (setInfo is null || setInfo.Beatmaps.Count() <= 1)
+                this.Exit();
+            var nextBeatmap = setInfo!.Beatmaps.First();
+
+            // Force a refresh of the beatmap (and beatmap set) so the `Change difficulty` list is also updated.
+            Beatmap.Value = beatmapManager.GetWorkingBeatmap(nextBeatmap, true);
+            SwitchToDifficulty(Beatmap.Value.BeatmapInfo);
```

Comment:
Sanity check here as well.

Reply:
These comments are unnecessary. We review every line.

Reply:
Apologies, spent too much time around rubber stamps 😅